src/systemctl/systemctl-edit.c

```diff
@@ -483,6 +483,18 @@ static int trim_edit_markers(const char *path) {
                 strshorten(contents_start, contents_end - contents_start);
 
         contents_start = strstrip(contents_start);
+        if (*contents_start && !endswith(contents_start, "\n")) {
```

Comment:
`*contents_start` is `char`, please do not use implicit cast to boolean. Maybe, better to use `isempty()`.